services/py-api/src/service/participants_service.py

```diff
@@ -35,10 +37,45 @@ async def _create_participant_and_team_in_transaction(
             return participant
 
         return Ok((participant.ok_value, team.ok_value))
+    
+    async def _create_random_participant_in_transaction(
+            self, input_data: ParticipantRequestBody, session: Optional[AsyncIOMotorClientSession] = None
+    ) -> Ok[Participant] | Err[DuplicateEmailError | DuplicateTeamNameError | Exception]:
+        participant = await self._participant_repo.create(input_data, session, team_id=None)
+        if is_err(participant):
+            return participant
+        
+        return Ok(participant.ok_value)
 
     async def register_admin_participant(
         self, input_data: ParticipantRequestBody
     ) -> Result[Tuple[Participant, Team], DuplicateEmailError | DuplicateTeamNameError | Exception]:
         # TODO: Add Capacity check 2
         result = await self._tx_manager.with_transaction(self._create_participant_and_team_in_transaction, input_data)
         return result
+    
+    async def capacity_check_one(self):
+        random_participants = ParticipantsRepository.get_number_random_participants()
+        registered_teams = TeamsRepository.get_number_registered_teams()
+        
+        hackathon_teams_cap = 20 #change later
+        participants_per_team = 5 #change later
+
+        anticipated_total_teams = registered_teams + ceil((random_participants + 1) / participants_per_team)
+
+        if anticipated_total_teams <= hackathon_teams_cap:
+            return True
+        else:
+            return False
+
+    #revise
+    async def insert_random_participant(self, input_data: ParticipantRequestBody
+    ) -> Ok[Tuple[Participant, Team]] | Err[DuplicateEmailError | DuplicateTeamNameError | Exception]:
+        
+        if ParticipantsRepository.capacity_check_one():
+           result = self._tx_manager.with_transaction(self._create_random_participant_in_transaction, input_data)
+           return result; 
+        else:
+           ParticipantHandlers.capacity_reached()
```

Comment:
Use something similar to this: 

```Python
async def register_admin_participant(self, input_data: ParticipantRequestBody) -> Result[
        Tuple[Participant, Team],
        DuplicateEmailError | DuplicateTeamNameError | HackathonCapacityExceededError | Exception,
    ]:

        # Capacity Check 2
        has_capacity = await self._check_capacity_register_admin_participant_case()
        if not has_capacity:
            return Err(HackathonCapacityExceededError(self._team_repo.MAX_NUMBER_OF_TEAMS_IN_HACKATHON))

        # Proceed with registration if there is capacity
        result = await self._tx_manager.with_transaction(self._create_participant_and_team_in_transaction, input_data)
        return result
```

With the new changes in main we have created a costum error type HackathonCapacityExceededError that you can throw when the check is not successful.

Reply:
Fixed in 1bec98abe406eea6382a5453ac98f3e900d3ddd4